proto/batch_plan.proto

```diff
@@ -66,6 +66,14 @@ message SourceNode {
   map<string, secret.SecretRef> secret_refs = 6;
 }
 
+message IcebergSourceNode {
+  uint32 source_id = 1;
+  repeated plan_common.ColumnCatalog columns = 2;
+  map<string, string> with_properties = 3;
+  repeated bytes split = 4;
+  map<string, secret.SecretRef> secret_refs = 5;
+}
```

Comment:
Let's change it to `IcebergScanNode` and keep only the necessary information like 'columns' and `with_properties`.